### PR TITLE
Mod picking upstream 2

### PR DESCRIPTION
**Pull request by TotalKrill**
Ci fixed  version of: #14541 

Upstream the remainder of bevy_picking_core and all of bevy_picking_input.

This work is intentionally nonfunctional and has minimal changes, but does compile. More work is necessary to replace bevy_eventlistener with propagating observers.

This work is being coordinated as part of "bevy_mod_picking upstream" working group. Come say hi on discord!

### CODE REVIEW

**Comment by SkiFire13** [comment]
From the previous PR https://github.com/bevyengine/bevy/pull/14541#issuecomment-2259047303

> Could you fix these two TODOs since you're adding the items the docs refer to?
>
> https://github.com/bevyengine/bevy/blob/ba09f354745b1c4420e6e316d80ac97c9eb37f56/crates/bevy_picking/src/lib.rs#L29-L32
>
> https://github.com/bevyengine/bevy/blob/ba09f354745b1c4420e6e316d80ac97c9eb37f56/crates/bevy_picking/src/lib.rs#L75-L78

**Comment by alice-i-cecile** [comment]
IMO we should revert the lock: it doesn't fix CI and a fix is due upstream anyways. Once that's done I'll get this merged.

**Comment by alice-i-cecile** [comment]
Once that reversion is done I think this is ready to merge :)

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_picking/src/lib.rs [hunk 1]
```diff
@@ -72,11 +73,7 @@ pub struct Pickable {
     ///
     /// Entities without the [`Pickable`] component will block by default.
     pub should_block_lower: bool,
-    // TODO: remove this allow after focus/hover is implemented in bevy_picking
-    #[allow(rustdoc::broken_intra_doc_links)]
-    /// Should this entity be added to the [`HoverMap`](focus::HoverMap) and thus emit events when
-    /// targeted?
-    ///
```
Comment (SkiFire13) [inline crates/bevy_picking/src/lib.rs:79]: I think this line was supposed to be part of the documentation and not the TODO to be removed.

**Pull request merged**